src/components/templates/PartyMap/components/Map/MapRoom.tsx

```diff
@@ -82,13 +87,23 @@ export const MapRoom: React.FC<MapRoomProps> = ({
     <div
       className={containerClasses}
       style={roomInlineStyles}
-      onClick={isUnclickable ? noop : selectRoomWithSound}
-      onMouseEnter={isUnclickable ? noop : handleRoomHovered}
-      onMouseLeave={isUnclickable ? noop : handleRoomUnhovered}
+      onClick={isUnclickable || isIframe ? noop : selectRoomWithSound}
+      onMouseEnter={isUnclickable || isIframe ? noop : handleRoomHovered}
+      onMouseLeave={isUnclickable || isIframe ? noop : handleRoomUnhovered}
     >
-      <img className="maproom__image" src={room.image_url} alt={room.title} />
+      {!isIframe ? (
+        <img className="maproom__image" src={room.image_url} alt={room.title} />
+      ) : (
+        <iframe
+          title={room.title}
+          id={`${room.type}-${room.title}`}
+          scrolling="no"
+          allow="autoplay"
+          src={room.url}
+        />
+      )}
```

Comment:
This feels a little messy.. I wonder if there is a better way we can handle this than having the ternary directly in the JSX like this? Though my brain isn't coming up with any great solutions right now.. @mike-lvov any thoughts?

Reply:
Why did we add the `id` prop to the `iframe`? We basically never use/need to use `id`'s in React.

---

I expect we're probably going to want to add a `className` to the iframe (probably `&__iframe`) or similar in the `.scss`) to apply some styling directly to this. At a minimum, setting the width/height to `100%` (so it fills it's parent container) seems like a decent idea?

---

It seems like `scrolling="no"` is deprecated. I expect this should also be controlled from CSS styles these days:

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#deprecated_attributes

Further reading:

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement

---

Instead of `allow="autoplay"`, we may/probably want to use our standard `IFRAME_ALLOW` from `src/settings`?

---

```suggestion
      {!isIframe ? (
        <img className="maproom__image" src={room.image_url} alt={room.title} />
      ) : (
        <iframe
          className="maproom__iframe"
          src={room.url}
          title={room.title}
          allow={IFRAME_ALLOW}
        />
      )}
```

Reply:
[OOS?] Eventually this (and all iframes in our codebase) should use a standard/generic 'Iframe wrapper component' that encapsulates our patterns/etc, but I don't think we have one in `staging` yet..

I believe @mike-lvov may have mentioned that he has one in a recent PR he has been working on potentially?

Reply:
done

Reply:
> This feels a little messy.. I wonder if there is a better way we can handle this than having the ternary directly in the JSX like this? Though my brain isn't coming up with any great solutions right now.. @mike-lvov any thoughts?

Not particularly, but I exchanged them in https://github.com/sparkletown/sparkle/pull/1427/commits/f5c38c702a23cf5d322715fa3c24eae1fb42e052

Reply:
I think my original pondering was around potentially using `switch`/`case` or similar, but seems fine enough as is for now given there are only 2 choices.

---

@mike-lvov Did you see the rest of my comments in this thread around deprecated features/etc and how they should be done through styling? It looks like you may have applied my suggested changes, or similar, but there aren't any styling changes made in the commit you linked?

Reply:
> @mike-lvov Did you see the rest of my comments in this thread around deprecated features/etc and how they should be done through styling? It looks like you may have applied my suggested changes, or similar, but there aren't any styling changes made in the commit you linked?

They were fixed before me. The latest version looks like this:
![image](https://user-images.githubusercontent.com/63194656/121259200-7953ba00-c8b8-11eb-92dc-f0a3217301ab.png)